task/control-plane-agent/src/mgs_gimlet.rs

```diff
@@ -235,6 +257,65 @@ impl MgsHandler {
         })
     }
 
+    pub(crate) fn uart_read(
+        &mut self,
+        data: Leased<idol_runtime::W, [u8]>,
+    ) -> Result<usize, ControlPlaneAgentError> {
+        // This function is only called by humility; switch control to it.
+        self.set_uart_client(UartClient::Humility)?;
+
+        let mut i = 0;
+        while i < data.len() {
+            let Some(b) = self.usart.from_rx.pop_front() else {
+                break;
+            };
+
+            if data.write_at(i, b).is_err() {
+                break;
+            }
+
+            i += 1;
+        }
+
+        if !self.usart.from_rx.is_full() {
```

Comment:
Should we also check that `i > 0` here?

Reply:
I don't think it's necessary to; if someone calls us and we have no buffered data or they gave us a 0-length lease, we'll (unnecessarily) re-enable the RX FIFO interrupt, but the same is true if someone calls us when `from_rx` wasn't full. The call to `enable_rx_interrupt()` is just a register write, not an IPC call to a separate driver task, so I didn't think I needed to worry too much about avoiding calling it unnecessarily. Do you think it's more confusing this way?

Reply:
Okay, I agree it's not necessary.  I'm on the fence whether it's more or less confusing...